patches/net/minecraft/server/packs/repository/PackRepository.java.patch

```diff
@@ -20,17 +20,17 @@
          for (RepositorySource repositorysource : this.sources) {
 -            repositorysource.loadPacks(p_143903_ -> map.put(p_143903_.getId(), p_143903_));
 +            Map<String, Pack> sourceMap = Maps.newTreeMap();
-+            repositorysource.loadPacks(p_143903_ -> sourceMap.put(p_143903_.getId(), p_143903_));
++            repositorysource.loadPacks(p_143903_ -> p_143903_.streamSelfAndChildren().forEach(p -> sourceMap.put(p.getId(), p)));
```

Comment:
Remember that the child packs don't get selected by ID, they get attached (ordered) to the parent pack. This change means that the child packs can be _queried_ by ID (turns out people need to do this), but will still be ordered properly because they're selected as children of the parent, not by ID, so it doesn't matter how they're ordered because they'll be ordered properly when selected that way.